src/card_operations/card_fetch_wallet_list.c

```diff
@@ -247,7 +246,6 @@ bool card_fetch_wallet_name(const uint8_t *wallet_id, char *wallet_name) {
   }
 
   if (0 == strlen(wallet_name)) {
-    delay_scr_init(ui_text_wallet_doesnt_exists_on_this_card, DELAY_TIME);
```

Comment:
Instead of removing the ui enirely, check if an abort is recieved and return without displaying UI

Reply:
I see, the p0 check is already added above, but i think it was failing as earlier `card_data` error hanling had bugs.